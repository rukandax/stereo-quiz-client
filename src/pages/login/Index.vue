<template>
  <div id="login-index">
    <div class="row vh-100 d-flex align-items-center justify-content-center">
      <div class="col-6">
        <div class="card shadow">
          <div class="card-body">
            <div class="form-group">
              <label for="nip">NIP</label>
              <input type="email" class="form-control" id="nip" placeholder="Masukan NIP">
            </div>
            <div class="form-group">
              <label for="password">Password</label>
              <input type="password" class="form-control" id="password" placeholder="Masukan Password">
            </div>
            <button class="btn btn-primary">Login</button>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script>
export default {
  methods: {
    handleLogin() {
      this.$router.push('/quiz');
    },
  },
};
</script>
