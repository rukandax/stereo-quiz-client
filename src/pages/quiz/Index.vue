<template>
  <div id="quiz-index" class="my-5">
    <div class="row">
      <div class="col-12">
        <div class="card shadow">
          <div class="card-body">
            <div class="action-button mb-4 d-flex justify-content-between align-items-center">
              <div class="btn btn-primary" data-toggle="modal" data-target="#modal-input-proktor">
                Kerjakan Assessment
              </div>
              <a href="#" class="profile">
                <img :src="`https://api.adorable.io/avatars/14/${user.email}.png`" class="img-thumbnail rounded mr-2">
                <span class="font-weight-bold">{{ user.name }}</span>
              </a>
            </div>

            <QuizHistory />
          </div>
        </div>
      </div>
    </div>

    <ModalInputProktor />
  </div>
</template>

<script>
import { mapState } from 'vuex';

import RequestMixin from '@/mixins/request-mixin';

import QuizHistory from '@/pages/quiz/components/QuizHistory.vue';
import ModalInputProktor from '@/pages/quiz/components/ModalInputProktor.vue';

export default {
  mixins: [
    RequestMixin,
  ],
  components: {
    QuizHistory,
    ModalInputProktor,
  },
  computed: {
    ...mapState(['user']),
  },
  mounted() {
    this.axiosGet('/me')
      .then(({ data }) => {
        this.$store.commit('updateUser', data);
      });
  },
};
</script>
