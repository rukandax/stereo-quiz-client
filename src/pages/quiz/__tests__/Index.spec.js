import { shallowMount } from '@vue/test-utils';

import Index from '@/pages/quiz/Index.vue';

describe('Pages Quiz Index', () => {
  let wrapper = null;

  beforeEach(() => {
    wrapper = shallowMount(Index);
  });

  afterEach(() => {
    wrapper = null;
  });

  it('Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Renders', () => {
    expect(wrapper).anything;
  });
});
