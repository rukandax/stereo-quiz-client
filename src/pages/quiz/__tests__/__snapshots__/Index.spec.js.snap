// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Quiz Index Snapshot 1`] = `
<div id="quiz-index" class="my-5">
  <div class="row">
    <div class="col-12">
      <div class="card shadow">
        <div class="card-body">
          <div class="action-button mb-4">
            <div data-toggle="modal" data-target="#modal-input-proktor" class="btn btn-primary">
              Kerjakan Assessment
            </div>
          </div>
          <quizhistory-stub></quizhistory-stub>
        </div>
      </div>
    </div>
  </div>
  <modalinputproktor-stub></modalinputproktor-stub>
</div>
`;
